sourceHandlers/gitHub.js

```diff
@@ -1,147 +1,136 @@
+      if (error == null) {
+        const sourceData = new SourceData();
+        // General info
+        const generalData = results[0];
+        sourceData.name = generalData.name;
+        sourceData.url = generalData.homepage || generalData.html_url;
+        sourceData.releaseDate = new Date(generalData.updated_at);
+        sourceData.description = generalData.description;
+
+        // Author Info
+        sourceData.authors = results[1];
+
+        // Version Data
+        const versions = results[2];
+        if (versions.length > 0) {
+          sourceData.version = versions[0].name || versions[1].tag_name;
+        }
+
+        callback(sourceData, []);
+      }
+      else {
+        callback(null, [error]);
+      }
+    });
+  }
+
+  /*
+   * Creates and sends a request for the GitHub API.
+   * @private
+   * @param {string} path - The path of the url.  This does not include the base path.  For example if you
+   * want to send a request to 'http://api.github.com/repos/apple/swift' use 'repos/apple/swift'.
+   * @param {Function} cb - The callback function. Follows the error/response parameter pattern.
+   * The response param will be json parsed object.
+   */
+  _sendApiRequest(path, cb) {
+    const options = {
+      url: this.url + path,
+      headers: {
+        'User-Agent': userAgent,
+      },
+    };
+
+    request(options, (error, response, body) => {
+      const parsedBody = (body != null) ? JSON.parse(body) : null;
+      if (error == null && response.statusCode !== 200) {
+        const errorMessage = `Received a ${response.statusCode} when making a request to ${options.url}`;
+        cb(new Error(errorMessage), parsedBody);
+      }
+      else {
+        cb(error, parsedBody);
+      }
+    });
+  }
+
+  /*
+   * Quries the the GitHub API to get the authors of a project
+   * @private
+   * @param {string[]} gitHubLogins - An array of GitHub username to query for information.
+   * @param {Function} callback - The callback function. Follows the error response parameter pattern.
+   * The response parameter is an array of Author objects
+   */
+  _fetchAuthors(gitHubLogins, callback) {
+    // Generate the requests that will feth the github user information
+    const userFetchOperations = gitHubLogins.map((obj, index) => (cb) => {
+      this._sendApiRequest(`users/${obj}`, (error, res) => {
+        cb(error, res);
+      });
+    });
+
+    // Execute those requests in parallel and generate the generic user objects
+    async.parallel(userFetchOperations, (error, results) => {
+      callback(error, results.map((obj) => {
+        const namePieces = (obj.name != null && typeof (obj.name) === 'string') ? obj.name.split(' ') : [];
+        return {
+          firstName: (namePieces.length > 0) ? namePieces[0] : null,
+          middleName: (namePieces.length > 2) ? namePieces[1] : null,
+          lastName: (namePieces.length > 2) ? namePieces[2] : namePieces[1],
```

Comment:
`obj.name` could be set to any string.

`lastName` will be undefined if only a single name is entered (e.g. "Prince").  `namePieces[1]` will be used since `namePieces.length` is 1 which is not greater than 2, but `namePieces[1]` is undefined.

`lastName` will be incorrect if more than three names are entered (e.g. "Darren Michael Luke O'Neill").  `namePieces[2]` will be used since `namePieces.length` is 4 which is greater than 2, but this will not be the true "last" name. In the example, "Luke" (`namePieces[2]`) will be used when the correct last name should be "O'Neill" (`namePieces[3]`).

To match as many names as possible as correctly as possible, I think we should use something along the lines of
``` js
const result = {};

result.firstName = (namePieces.length > 1) ? namePieces[0] : null;

if (namePieces.length > 2) {
    result.middleName = (() => {
        var mName = "";
        namePieces.forEach((namePiece, i) => {
            if (i > 0 && i < namePieces.length - 1) {
                mName += " " + namePiece;
            }
        });
        return mName;
    })();
}
else {
    result.middleName == null;
}

result.lastName = (namePieces.length > 0) ? namePieces[namePieces.length - 1] : null;

result.email = obj.email;

return returnObject;
```

Examples:

"Prince"
- `firstName`: null
- `middleName`: null
- `lastName`: "Prince"

"Colin O'Neill"
- `firstName`: "Colin"
- `middleName`: null
- `lastName`: "O'Neill"

"Eric David Lee"
- `firstName`: "Eric"
- `middleName`: "David"
- `lastName`: "Lee"

"Darren Michael Luke O'Neill"
- `firstName`: "Darren"
- `middleName`: "Michael Luke"
- `lastName`: "O'Neill"

""
- `firstName`: null
- `middleName`: null
- `lastName`: null